diff --git a/Makefile b/Makefile
index 6d4a44f..f244a14 100644
--- a/Makefile
+++ b/Makefile
@@ -901,7 +901,7 @@ clobber:	tidy
 		-o -name '*.bin' -o -name u-boot.img \) \
 		-print0 | xargs -0 rm -f
 	@rm -f $(OBJS) $(obj)*.bak $(obj)ctags $(obj)etags $(obj)TAGS \
-		$(obj)cscope.* $(obj)*.*~
+		$(obj)*.*~
 	@rm -f $(obj)u-boot $(obj)u-boot.map $(obj)u-boot.hex $(ALL-y)
 	@rm -f $(obj)u-boot.kwb
 	@rm -f $(obj)u-boot.pbl
diff --git a/drivers/scboot/otp.c b/drivers/scboot/otp.c
index 1d2c012..8aa6af3 100644
--- a/drivers/scboot/otp.c
+++ b/drivers/scboot/otp.c
@@ -31,7 +31,7 @@ int get_rsakeylen(void)
 static void gpio_output_value(int gpio, int value)
 {
 	gpio_direction_output(gpio, value);
-	mdelay(10);		/* wait for EFUSE IO power for mdelay(10). */
+	mdelay(2);		/* wait for EFUSE IO power for mdelay(10). */
 }
 
 
@@ -39,9 +39,11 @@ static int efuse_update_state(void)
 {
 	volatile unsigned int *reg_ctrl = (volatile unsigned int *)EFUSE_REG_CTRL;
 	volatile unsigned int *reg_stat = (volatile unsigned int *)EFUSE_REG_STAT;
+	*reg_stat = 0;			      /* clear WR_DONE RD_DONE */
 	*reg_ctrl = (EFUSE_ADDR_PROT << EFUSE_REGOFF_CRTL_ADDR) | (0x1 << EFUSE_REGOFF_CRTL_LENG) | EFUSE_REG_CTRL_RDEN;
 	while(!(*reg_stat & EFUSE_REG_STAT_RDDONE));
-	printf("xxxxxxx state updated: %x\n", *reg_stat);
+	*reg_stat = 0;			      /* clear WR_DONE RD_DONE */
+	printf("%s() updated EFUSE_STAT: %x\n", __func__, *reg_stat);
 }
 
 static int efuse_config(void)
@@ -76,14 +78,14 @@ static int efuse_config(void)
 		int apll_o = ((*reg_cpapcr & 0x3<<16)>>16) + 1;
 
 		pll = 24*apll_m/(apll_n * apll_o);
-//		printf(" xxxx AHB2 select APLL : %d\n", pll);
+		//printf(" xxxx AHB2 select APLL : %d\n", pll);
 	} else if(sel_a == 2) {
 		int mpll_m = ((*reg_cpmpcr & 0x7f<<24)>>24) + 1;
 		int mpll_n = ((*reg_cpmpcr & 0x1f<<18)>>18) + 1;
 		int mpll_o = ((*reg_cpmpcr & 0x3<<16)>>16) + 1;
 
 		pll = 24*mpll_m/(mpll_n*mpll_o);
-//		printf(" xxxx AHB2 select MPLL : %d\n", pll);
+		//printf(" xxxx AHB2 select MPLL : %d\n", pll);
 	}
 
 	int ahb2 = pll/h2div;
@@ -128,14 +130,16 @@ static int efuse_config(void)
 	//rd_adj = 100;
 	//rd_strobe = 100;
 	*reg_cfg = (rd_adj << 19) | (rd_strobe << 16) | (wr_adj<<12) | wr_strobe;
-//	printf("xxxxxxx reg_cfg = %x\n",*reg_cfg);
-//	printf("xxxxxxx mpll = %d\n",pll);
-//	printf("xxxxxxx ahb2 = %d\n",ahb2);
+	printf("xxxxxxx efuse reg_cfg = %x\n",*reg_cfg);
+	printf("xxxxxxx mpll = %d\n",pll);
+	printf("xxxxxxx ahb2 = %d\n",ahb2);
 
 }
 
 void otp_init(void)
 {
+	//printf("xxxxxxxxxxx func : %s() debug_args->efuse_gpio=%d\n",__func__, debug_args->efuse_gpio);
+
 	gpio_output_value(debug_args->efuse_gpio, 1);
 	efuse_config();
 	efuse_update_state();
@@ -149,9 +153,11 @@ int cpu_wtotp(int opera)
 	volatile struct sc_args *args;
 	args = (volatile struct sc_args *)GET_SC_ARGS();
 	volatile unsigned int *reg_ctrl = (volatile unsigned int *)EFUSE_REG_CTRL;
+	volatile unsigned int *reg_stat = (volatile unsigned int *)EFUSE_REG_STAT;
 
 //	printf("xxxxxxxxxxx func : %s\n",__func__);
 
+	*reg_stat = 0;			      /* clear WR_DONE RD_DONE */
 	*reg_ctrl |= EFUSE_REG_CTRL_PGEN; /*pg en*/
 	gpio_output_value(debug_args->efuse_gpio, 0);
 
@@ -160,6 +166,8 @@ int cpu_wtotp(int opera)
 
 	gpio_output_value(debug_args->efuse_gpio, 1);
 	*reg_ctrl &= ~EFUSE_REG_CTRL_PGEN;
+	*reg_stat = 0;			      /* clear WR_DONE RD_DONE */
+	mdelay(2);		/* mdelay 2ms after clear CTRL_PGEN, waiting for AVDEFUSE_2V5 down. */
 
 	if (*(volatile unsigned int *)(MCU_TCSM_RETVAL) != SC_ERR_SUCC) {
 		printf("write otp err, ret val %x\n",*(volatile unsigned int *)(MCU_TCSM_RETVAL));
@@ -177,6 +185,7 @@ int cpu_burn_rckey(void)
 	volatile struct sc_args *args;
 	args = (volatile struct sc_args *)GET_SC_ARGS();
 	volatile unsigned int *reg_ctrl = (volatile unsigned int *)EFUSE_REG_CTRL;
+	volatile unsigned int *reg_stat = (volatile unsigned int *)EFUSE_REG_STAT;
 
 //	printf("xxxxxxxxxxx func : %s\n",__func__);
 	return 0;
@@ -186,6 +195,7 @@ int cpu_burn_rckey(void)
 	if(cpu_get_rn() < 0)
 		return -ESEC;
 
+	*reg_stat = 0;			      /* clear WR_DONE RD_DONE */
 	*reg_ctrl |= EFUSE_REG_CTRL_PGEN; /*pg en*/
 	gpio_output_value(debug_args->efuse_gpio, 0);
 
@@ -193,6 +203,8 @@ int cpu_burn_rckey(void)
 
 	gpio_output_value(debug_args->efuse_gpio, 1);
 	*reg_ctrl &= ~EFUSE_REG_CTRL_PGEN;
+	*reg_stat = 0;			      /* clear WR_DONE RD_DONE */
+	mdelay(2);		/* mdelay 2ms after clear CTRL_PGEN, waiting for AVDEFUSE_2V5 down. */
 
 	if (*(volatile unsigned int *)(MCU_TCSM_RETVAL) != SC_ERR_SUCC) {
 		printf("burn rckey err, ret val %x\n",*(volatile unsigned int *)(MCU_TCSM_RETVAL));
@@ -335,33 +347,48 @@ int cpu_burn_secboot_enable(void)
 	volatile unsigned int *reg_stat = (volatile unsigned int *)EFUSE_REG_STAT;
 	volatile unsigned int *reg_data1 = (volatile unsigned int *)EFUSE_REG_DAT1;
 
+	*reg_data1 = (1 << EFUSE_PTCOFF_SEC); /* program security boot enable */
+	//*reg_ctrl = ~(0x7f << EFUSE_REGOFF_CRTL_ADDR | 0x1f << EFUSE_REGOFF_CRTL_LENG); /* clean address ,length*/
+	*reg_stat = 0;			      /* clear WR_DONE RD_DONE */
+	*reg_ctrl = 0;
+	*reg_ctrl |= EFUSE_ADDR_PROT << EFUSE_REGOFF_CRTL_ADDR; /* set address, length(0+1) */
+
 	*reg_ctrl |= EFUSE_REG_CTRL_PGEN; /*pg en*/
 	gpio_output_value(debug_args->efuse_gpio, 0);
 
-	*reg_data1 = (1 << EFUSE_PTCOFF_SEC); /* security boot enable, security boot enable protected */
-	*reg_ctrl &= ~(0x7f << EFUSE_REGOFF_CRTL_ADDR | 0x1f << EFUSE_REGOFF_CRTL_LENG); /* clean address ,length*/
-	*reg_ctrl |= EFUSE_ADDR_PROT << EFUSE_REGOFF_CRTL_ADDR; /* set address */
 	*reg_ctrl |= EFUSE_REG_CTRL_WTEN; /*write en*/
 
 	while(!(*reg_stat & EFUSE_REG_STAT_WTDONE));
 
 	gpio_output_value(debug_args->efuse_gpio, 1);
-	*reg_ctrl &= ~EFUSE_REG_CTRL_PGEN;
+	//*reg_ctrl &= ~EFUSE_REG_CTRL_PGEN;
+	*reg_ctrl = 0;
+	*reg_stat = 0;			      /* clear WR_DONE RD_DONE */
+	mdelay(2);		/* mdelay 2ms after clear CTRL_PGEN, waiting for AVDEFUSE_2V5 down. */
 
 	efuse_update_state();
 
+	if(!(*reg_stat & EFUSE_REG_STAT_SCBT_EN)) {
+		printf("%s() security boot enable write failed!, reg_stat=%x\n", __func__, *reg_stat);
+		return -1;
+	}
+
+	*reg_data1 = (1 << EFUSE_PTCOFF_SCB); /* program security boot enable protected */
+	*reg_stat = 0;			      /* clear WR_DONE RD_DONE */
+	*reg_ctrl = 0;
+	*reg_ctrl |= EFUSE_ADDR_PROT << EFUSE_REGOFF_CRTL_ADDR; /* set address, length(0+1) */
 	*reg_ctrl |= EFUSE_REG_CTRL_PGEN; /*pg en*/
 	gpio_output_value(debug_args->efuse_gpio, 0);
 
-	*reg_data1 = (1 << EFUSE_PTCOFF_SCB); /* security boot enable, security boot enable protected */
-	*reg_ctrl &= ~(0x7f << EFUSE_REGOFF_CRTL_ADDR | 0x1f << EFUSE_REGOFF_CRTL_LENG); /* clean address ,length*/
-	*reg_ctrl |= EFUSE_ADDR_PROT << EFUSE_REGOFF_CRTL_ADDR; /* set address */
 	*reg_ctrl |= EFUSE_REG_CTRL_WTEN; /*write en*/
 
 	while(!(*reg_stat & EFUSE_REG_STAT_WTDONE));
 
 	gpio_output_value(debug_args->efuse_gpio, 1);
-	*reg_ctrl &= ~EFUSE_REG_CTRL_PGEN;
+	//*reg_ctrl &= ~EFUSE_REG_CTRL_PGEN;
+	*reg_ctrl = 0;
+	*reg_stat = 0;			      /* clear WR_DONE RD_DONE */
+	mdelay(2);		/* mdelay 2ms after clear CTRL_PGEN, waiting for AVDEFUSE_2V5 down. */
 
 	efuse_update_state();
 
diff --git a/drivers/scboot/otp.h b/drivers/scboot/otp.h
index 68d40cb..c475749 100644
--- a/drivers/scboot/otp.h
+++ b/drivers/scboot/otp.h
@@ -30,6 +30,7 @@
 
 #define EFUSE_REG_STAT_RDDONE 0x1
 #define EFUSE_REG_STAT_WTDONE (0x1 << 1)
+#define EFUSE_REG_STAT_SCBT_EN (0x1 << 8)
 #define EFUSE_REG_STAT 0xb3540008
 
 #define EFUSE_REG_DAT1 0xb354000C
